File: LeetCode/Array/Golang/Leetcode1996-游戏中的弱角色.go
package sbgo

import "sort"

func numberOfWeakCharacters(ps [][]int) int {
    sort.Slice(ps,func(l, r int) bool {
        p,q := ps[l],ps[r]
        return p[0] > q[0] || (p[0] == q[0] && p[1] < q[1]) 
    })
    ans,mx := 0,0
    for _,p := range ps{
        se := p[1]
        if se < mx {
            ans++
        }else if se > mx{
            mx = se
        }
    }
    return ans
}

File: LeetCode/Array/Python/Leetcode1996-游戏中的弱角色.py
class Solution:
    def numberOfWeakCharacters(self, ps: List[List[int]]) -> int:
        ps.sort(key = lambda x : (-x[0],x[1]))
        ans,mx = 0,0
        for _,se in ps:
            if se < mx: ans += 1
            else: mx = max(mx,se)
        return ans